Mathlib/AlgebraicTopology/Quasicategory/StrictSegal.lean

```diff
@@ -39,64 +39,54 @@ instance quasicategory {X : SSet.{u}} [StrictSegal X] : Quasicategory X := by
   · rw [← spine_arrow, spine_δ_arrow_lt _ hlt]
     dsimp only [Path.map, spine_arrow, Fin.coe_eq_castSucc]
     apply congr_arg
-    simp only [horn, horn.spineId, stdSimplex, uliftFunctor, Functor.comp_obj,
-      yoneda_obj_obj, whiskering_obj_obj_map, uliftFunctor_map, yoneda_obj_map,
-      stdSimplex.objEquiv, Equiv.ulift, Equiv.coe_fn_symm_mk,
-      Quiver.Hom.unop_op, horn.face_coe, Subtype.mk.injEq]
-    rw [mkOfSucc_δ_lt hlt]
+    apply Subtype.ext
+    dsimp [horn.face, CosimplicialObject.δ]
+    rw [Subcomplex.yonedaEquiv_coe, Subpresheaf.lift_ι, stdSimplex.map_apply,
+      Quiver.Hom.unop_op, stdSimplex.yonedaEquiv_map, Equiv.apply_symm_apply,
+      mkOfSucc_δ_lt hlt]
     rfl
   · rw [← spine_arrow, spine_δ_arrow_gt _ hgt]
     dsimp only [Path.map, spine_arrow, Fin.coe_eq_castSucc]
     apply congr_arg
-    simp only [horn, horn.spineId, stdSimplex, uliftFunctor, Functor.comp_obj,
-      yoneda_obj_obj, whiskering_obj_obj_map, uliftFunctor_map, yoneda_obj_map,
-      stdSimplex.objEquiv, Equiv.ulift, Equiv.coe_fn_symm_mk,
-      Quiver.Hom.unop_op, horn.face_coe, Subtype.mk.injEq]
-    rw [mkOfSucc_δ_gt hgt]
+    apply Subtype.ext
+    dsimp [horn.face, CosimplicialObject.δ]
+    rw [Subcomplex.yonedaEquiv_coe, Subpresheaf.lift_ι, stdSimplex.map_apply,
+      Quiver.Hom.unop_op, stdSimplex.yonedaEquiv_map, Equiv.apply_symm_apply,
+      mkOfSucc_δ_gt hgt]
     rfl
-  · /- The only inner horn of `Δ[2]` does not contain the diagonal edge. -/
-    have hn0 : n ≠ 0 := by
-      rintro rfl
+  · obtain _ | n := n
+    · /- The only inner horn of `Δ[2]` does not contain the diagonal edge. -/
       obtain rfl : k = 0 := by omega
       fin_cases i <;> contradiction
-    /- We construct the triangle in the standard simplex as a 2-simplex in
-    the horn. While the triangle is not contained in the inner horn `Λ[2, 1]`,
-    we can inhabit `Λ[n + 2, i] _[2]` by induction on `n`. -/
-    let triangle : Λ[n + 2, i] _[2] := by
-      cases n with
-      | zero => contradiction
-      | succ _ => exact horn.primitiveTriangle i h₀ hₙ k (by omega)
-    /- The interval spanning from `k` to `k + 2` is equivalently the spine
-    of the triangle with vertices `k`, `k + 1`, and `k + 2`. -/
-    have hi : ((horn.spineId i h₀ hₙ).map σ₀).interval k 2 (by omega) =
-        X.spine 2 (σ₀.app _ triangle) := by
-      ext m
-      dsimp [spine_arrow, Path.interval, Path.map]
-      rw [← types_comp_apply (σ₀.app _) (X.map _), ← σ₀.naturality]
+    · /- We construct the triangle in the standard simplex as a 2-simplex in
+      the horn. While the triangle is not contained in the inner horn `Λ[2, 1]`,
+      we can inhabit `Λ[n + 3, i] _[2]` by induction on `n`. -/
```

Comment:
```suggestion
      it suffices to inhabit `Λ[n + 3, i] _[2]`. -/
```
It appears the rather awkward induction argument is no longer needed, so this removes all mention of it.

Reply:
Thanks!